be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/GetMessagesByIdResponseHandler.scala

```diff
@@ -0,0 +1,66 @@
+package ch.epfl.pop.pubsub.graph.handlers
+
+import akka.NotUsed
+import akka.actor.ActorRef
+import akka.stream.scaladsl.{Flow, GraphDSL, Merge, Partition, Sink}
+import ch.epfl.pop.model.network.{JsonRpcRequest, JsonRpcResponse}
+import ch.epfl.pop.pubsub.graph.{ErrorCodes, GraphMessage, PipelineError}
+import ch.epfl.pop.pubsub.{AskPatternConstants, ClientActor, PubSubMediator}
+import ch.epfl.pop.storage.DbActor
+import akka.stream.FlowShape
+
+/** This object's job is to handle responses it receives from other servers after sending a heartbeat. When receiving the missing messages, the server's job is to write them on the data base.
+  */
+object GetMessagesByIdResponseHandler extends AskPatternConstants {
+
+  def graph(dbActorRef: ActorRef): Flow[GraphMessage, GraphMessage, NotUsed] = Flow.fromGraph(GraphDSL.create() {
+    implicit builder: GraphDSL.Builder[NotUsed] =>
+      {
+        import GraphDSL.Implicits._
+
+        /* partitioner port numbers */
+        val portPipelineError = 0
+        val portResponseHandler = 1
+        val totalPorts = 2
+
+        /* building blocks */
+        val handlerPartitioner = builder.add(Partition[GraphMessage](
+          totalPorts,
+          {
+            case Right(jsonRpcMessage: JsonRpcResponse) => jsonRpcMessage.result match {
+                case Some(result) => result.resultMap match {
+                    case Some(map) => portResponseHandler
```

Comment:
Since `map` is unused: 
```suggestion
                    case Some(_) => portResponseHandler
```